views/frontend/app.blade.php

```diff
@@ -64,5 +64,7 @@
     @endif
 
     {!! $foot !!}
+
+    {!! array_get($forum, 'attributes.footerHtml') !!}
```

Comment:
Instead of dumping this out here as well, let's keep the view dumb, and instead append this onto `$foot` when [preparing the view](https://github.com/flarum/core/blob/63be95fb8a1afc2c9633268701c6b45021a00f6a/src/Frontend/FrontendView.php#L307).

Reply:
Hi, @franzliedke .

I realize the commit you pointed are newer than mine.
So i tried a rebase to the updated master branch before make another changes.

The rebase was successful, but i got an error when accessing the flarum on my machine.

It's shows:
```
Fatal error: Uncaught Flarum\Http\Exception\RouteNotFoundException in C:\Users\afrastgeek\Projects\flarum\vendor\flarum\core\src\Http\Middleware\DispatchRoute.php:62 Stack trace: #0 C:\Users\afrastgeek\Projects\flarum\vendor\zendframework\zend-stratigility\src\Dispatch.php(212): Flarum\Http\Middleware\DispatchRoute->__invoke(Object(Zend\Stratigility\Http\Request), Object(Zend\Stratigility\Http\Response), Object(Zend\Stratigility\Next)) #1 C:\Users\afrastgeek\Projects\flarum\vendor\zendframework\zend-stratigility\src\Dispatch.php(88): Zend\Stratigility\Dispatch->dispatchCallableMiddleware(Object(Flarum\Http\Middleware\DispatchRoute), Object(Zend\Stratigility\Next), Object(Zend\Stratigility\Http\Request), Object(Zend\Stratigility\Http\Response), NULL) #2 C:\Users\afrastgeek\Projects\flarum\vendor\zendframework\zend-stratigility\src\Next.php(160): Zend\Stratigility\Dispatch->__invoke(Object(Zend\Stratigility\Route), NULL, Object(Zend\Stratigility\Http\Request), Object(Zend\Stratigility\Http\Response), Object(Zend\Stratigilit in C:\Users\afrastgeek\Projects\flarum\vendor\flarum\core\src\Http\Middleware\DispatchRoute.php on line 62
```

I'm sorry but I don't know what to do.

Reply:
Try changing your Nginx config from

~~~
location / { try_files $uri $uri/ /index.php?$query_string; }
location /api { try_files $uri $uri/ /api.php?$query_string; }
location /admin { try_files $uri $uri/ /admin.php?$query_string; }
~~~

to

~~~
location / { try_files $uri $uri/ /index.php?$query_string; }
~~~

Also, make sure that flarum/flarum is up-to-date as well, and that - again in the Nginx configuration - `root` points to the `public` directory.

That might make it work. :confused:

Reply:
Thank you,

I use apache instead of nginx. And from your clue above, i tried changing redirection config but no luck.

Then i create a `http://flarum.localhost` vhost with root directory to  `C:\Users\afrastgeek\Projects\flarum\public` instead of accessing flarum from `http://localhost/flarum/public` and it works.

Creating a symlink of `htdocs` to `C:\Users\afrastgeek\Projects\flarum\public` also works.

Seems like it must be on a root directory, and can't tolerate placing it in a directory.

But nonetheless, thank you for your response.